pyanaconda/modules/storage/devicetree/viewer.py

```diff
@@ -519,15 +519,23 @@ def _get_windows_data(self):
         windows_data.os_name = "Windows"
         windows_data.devices = []
 
+        efi_partition = None
         for blivet_device in self.storage.devicetree.devices:
             if not isinstance(blivet_device, PartitionDevice):
                 continue
 
             device = self._get_device(blivet_device.name)
+            if str(device.part_type_uuid) == EFI_PARTITION_TYPE:
+                efi_partition = device
+                continue
+
             if str(device.part_type_uuid) in WINDOWS_PARTITION_TYPES:
                 windows_data.devices.append(device.name)
 
         if len(windows_data.devices) > 0:
+            if efi_partition is not None:
+                windows_data.devices.append(efi_partition.name)
```

Comment:
If this doesn't necessarily mean that the EFI is part of the Windows then we shouldn't use `windows_data` as the variable?

Reply:
Well if there is Windows, and there is an EFI partition and you delete it windows wont boot. So you should account it in the Windows devices. 
Similarly to how Linux devices contain /boot.

Reply:
The point of this commit is that if you find an EFI partition you should not directly assume there is windows. If there is windows the EFI partition belongs to that (as well)

Reply:
I see. I guess I misunderstood the code. Thanks for explanation.